ecl/hql/hqlgram.y

```diff
@@ -11555,7 +11557,7 @@ nonDatasetExpr
     : expression            
                         {
                             node_operator op = $1.getOperator();
-                            if (op == no_sortlist) //|| (op == no_list))
+                            if (op == no_sortlist || op==no_dedup) //|| (op == no_list))
```

Comment:
I think will need @ghalliday to comment as to whether sortListExpr and change here to allow no_dedup in nonDatasetExpr is good or not.